source/routing/raptor_api.cpp

```diff
@@ -1167,74 +1167,135 @@ void make_response(navitia::PbCreator& pb_creator,
+        // Compute start time and Bound
+        DateTime request_date_secs = to_datetime(datetime.first, raptor.data);
+
+        if(max_duration != DateTimeUtils::inf) {
+            if (clockwise) {
+                bound = request_date_secs + max_duration;
+            } else {
+                bound = request_date_secs > max_duration ? request_date_secs - max_duration : 0;
             }
-            std::vector<Path> tmp = raptor.compute_all(
-                *departures, *destinations, init_dt, rt_level, transfer_penalty, bound, max_transfers,
+        }
+
+        // Call raptor
+        // note : Loop is for min_nb_journeys options
+        //
+        // min_nb_journeys options :
+        // Compute several loop until the number of journeys >= min_nb_journeys
+        // For each step, we fing best pathes,
+        // start to the latest departure + 1 (clockwise) or the latest arrival - 1.
+        // If Raptor does not return anything, we stop.
+        // If the number of Path finds is greater or equal than min_nb_journeys, we stop.
+        JourneySet journeys;
+        uint32_t nb_try = 0;
+        do {
+            auto raptor_journeys = raptor.compute_all_journeys(
+                *departures, *destinations, request_date_secs, rt_level, transfer_penalty, bound, max_transfers,
                 accessibilite_params, forbidden, allowed, clockwise, direct_path_dur,
                 max_extra_second_pass);
-            LOG4CPLUS_DEBUG(logger, "raptor found " << tmp.size() << " solutions");
 
+            LOG4CPLUS_DEBUG(logger, "raptor found " << raptor_journeys.size() << " solutions");
 
-            // Lorsqu'on demande qu'un seul horaire, on garde tous les résultas
-            if(datetimes.size() == 1) {
-                pathes = tmp;
-                for(auto & path : pathes) {
-                    path.request_time = datetime;
-                }
-            } else if(!tmp.empty()) {
-                // Lorsqu'on demande plusieurs horaires, on garde que l'arrivée au plus tôt / départ au plus tard
-                tmp.back().request_time = datetime;
-                pathes.push_back(tmp.back());
-                bound = to_datetime(tmp.back().items.back().arrival, raptor.data);
-            } else // Lorsqu'on demande plusieurs horaires, et qu'il n'y a pas de résultat, on retourne un itinéraire vide
-                pathes.push_back(Path());
+            // Remove direct path
+            filter_direct_path(raptor_journeys);
+
+            if (raptor_journeys.empty())
+                break;
+
+            // Prepare next call for raptor with min_nb_journeys option
+            request_date_secs = prepare_next_call_for_raptor(raptor_journeys, clockwise);
+
+            // filter the similar journeys
+            for(const auto & journey : raptor_journeys) {
+                journeys.insert(journey);
+            }
+
+            nb_try++;
+
+        } while (( journeys.size() < min_nb_journeys) && (nb_try < MAX_NB_RAPTOR_CALL));
```

Comment:
100 try seems to be too much, in jormun  `MAX_NB_RAPTOR_CALL` is equal to `min_nb_journeys`, it's not that bad but I think we need to set lower and upper limit if we choose to do it this way.

Reply:
MAX_NB_RAPTOR_CALL = 100 is temporary.
I think is in the next JIRA ticket to add a flexible limit : https://jira.kisio.org/secure/RapidBoard.jspa?rapidView=111&projectKey=NAVP&view=planning&selectedIssue=NAVP-859